### ISSUE TITLE
Automatic Genre Tagging for Movies and Shows

### ISSUE THREAD

**Issue by jd4883** [jd4883/plex-linker-beta#18]
This is kind of to go with a second program I want to write as a companion to the linker. Using the API for Radarr/or Sonarr I want to pull all genre information and tag it directly to the show or movie. The reason for this is that my setup I use multiple subfolders for TV and movies to separate what disks I assign the content to. What I hope to do is use all tags and create a comparison between the folder names and make decisions on an appropriate root folder is to assign the show / movie. I plan to do this as a separate program to leverage the tags since this does not really apply to everyone's setup. 

I have not figured out how to manipulate tags using the sonarr or radarr endpoints yet but once determined I'll put a POC together for this secondary choice as a separate plex tool. Essentially this is what I want this programs workflow to do:
- Add any missing tags for each show / movie using the sonarr / radarr API's
- create a table of all genre names
- create a table of all movie folder names
- create a folder of all show folder names
- use a yaml to map what is considered a match along with a default path (staging in my setup)
- in the yaml, define a capacity where the root folder is considered too full (default this to 500 gigabytes)
- check available space, if capacity is within the threshold adjust the root folder for the item to the most appropriate option
- basically this should just be a lot of conditionals that do nothing if none are met but if they are met it can adjust the root folder, giving better spacial distribution and more accurate classifications of where movies / shows will live within the library. I've done a ton of this manually and automating this process should be pretty trivial if I can add the tags in automatically as identifiers.
- Once the root folder is changed, a tag "preferred root set" should be used to prevent future parsing of this item
